src/plugins/vis_type_timelion/server/lib/reduce.js

```diff
@@ -48,16 +49,17 @@ async function pairwiseReduce(left, right, fn) {
   });
 
   // pairwise reduce seriesLists
-  const pairwiseSeriesList = { type: 'seriesList', list: [] };
-  left.list.forEach(async (leftSeries) => {
-    const first = { type: 'seriesList', list: [leftSeries] };
-    const second = { type: 'seriesList', list: [indexedList[leftSeries[pairwiseField]]] };
-    const reducedSeriesList = await reduce([first, second], fn);
-    const reducedSeries = reducedSeriesList.list[0];
-    reducedSeries.label = leftSeries[pairwiseField];
-    pairwiseSeriesList.list.push(reducedSeries);
-  });
-  return pairwiseSeriesList;
+  return {
+    type: 'seriesList',
+    list: await asyncMap(left.list, async (leftSeries) => {
+      const first = { type: 'seriesList', list: [leftSeries] };
+      const second = { type: 'seriesList', list: [indexedList[leftSeries[pairwiseField]]] };
+      const reducedSeriesList = await reduce([first, second], fn);
+      const reducedSeries = reducedSeriesList.list[0];
+      reducedSeries.label = leftSeries[pairwiseField];
+      return reducedSeries;
+    }),
```

Comment:
The original implementation was producing a result list ordered by the speed of the `reduce()` function, which I assume was unintended and the `list` will now always mirror the order of `left.list`